modules/util/util.js

```diff
@@ -542,3 +542,45 @@ export function utilUnicodeCharsCount(str) {
 export function utilUnicodeCharsTruncated(str, limit) {
     return Array.from(str).slice(0, limit).join('');
 }
+
+// Returns whether two date component arrays represent the same date.
+// A date component array contains the full date, followed by year, month, day.
+// Dates are only compared to the lowest common precision.
+function isSameDate(date1, date2) {
+    return isBefore(date1, date2) && isAfter(date1, date2);
+}
+
+function isBefore(date1, date2) {
+    if (date1[1] > date2[1]) return false;
+    if (date1[2] && date2[2] && date1[2] > date2[2]) return false;
+    if (date1[3] && date2[3] && date1[3] > date2[3]) return false;
```

Comment:
Dates are only compared to the lowest common precision. If one feature ends on “1970” and another begins on “1970-01-01”, the two features are not considered to overlap temporally. This contrasts with the tile server, which rounds `start_date` to down to the beginning of the month or year and `end_date` up to the end of the month or year.